### PR TITLE
All: compile with clang and remove warning

### PR DESCRIPTION
**Pull request by pbougue**
remove clang warnings except from third_party and protobuf

### CODE REVIEW

File: source/cities/cities.cpp [hunk 1]
```diff
@@ -301,9 +301,17 @@ void OSMRelation::build_polygon(OSMCache& cache, std::set<u_int64_t> explored_id
         const auto front = tmp_polygon.outer().front();
         const auto back = tmp_polygon.outer().back();
         // This should not happen, but does some time
+
+#ifdef __clang__
```
Comment (TeXitoi) [inline source/cities/cities.cpp]: If it works under gcc, please remove the ifdef.

File: source/ed/connectors/gtfs_parser.cpp [hunk 2]
```diff
@@ -78,7 +78,7 @@ template <typename T>
 void add_gtfs_comment(GtfsData& gtfs_data, Data& data, const T& obj, const std::string& comment) {
     std::string& comment_id = gtfs_data.comments_id_map[comment];
     if (comment_id.empty()) {
-        comment_id = "comment__" + gtfs_data.comments_id_map.size();
```
Comment (TeXitoi) [inline source/ed/connectors/gtfs_parser.cpp:81]: outch!

File: source/ed/ed_reader.cpp [hunk 3]
```diff
@@ -1389,7 +1389,8 @@ void EdReader::fill_graph(navitia::type::Data& data, pqxx::work& work) {
             e.duration = navitia::seconds(len / ng::default_speed[nt::Mode_e::Walking]);
 
             // overflow check since we want to store that on a int32
-            if (e.duration.total_seconds() == std::floor(len / ng::default_speed[nt::Mode_e::Walking])) {
+            int dur = boost::lexical_cast<int>(std::floor(len / ng::default_speed[nt::Mode_e::Walking]));
+            if (e.duration.total_seconds() == dur) {
```
Comment (TeXitoi) [inline source/ed/ed_reader.cpp]: I think we don't want that.

File: source/ed/connectors/gtfs_parser.cpp (hunk 2 above)
Comment (pbougue) [inline source/ed/connectors/gtfs_parser.cpp:81]: right? :)

File: source/ed/ed_reader.cpp [hunk 4]
```diff
@@ -1352,6 +1352,19 @@ void EdReader::fill_vertex(navitia::type::Data& data, pqxx::work& work) {
...
+boost::optional<navitia::time_res_traits::sec_type>
+EdReader::get_duration (nt::Mode_e mode, float len, uint64_t source, uint64_t target) {
+    try {
+        // overflow check since we want to store that on a int32
+        return  boost::lexical_cast<navitia::time_res_traits::sec_type>(
+                                    std::floor(len / ng::default_speed[mode]));
+    } catch (const boost::bad_lexical_cast&) {
+        LOG4CPLUS_WARN(log, "edge length overflow for " << mode << " for source " << source
+                            << " target " << target << " length: " << len << ", we ignore this edge");
+        return false;
```
Comment (TeXitoi) [inline source/ed/ed_reader.cpp]: `return boost::none`

**Comment by TeXitoi** [comment]
OK for me.

**Comment by pbougue** [comment]
Jenkins, test this please.

**Comment by pbougue** [comment]
Jenkins, test this please.

**Comment by pbougue** [comment]
Jenkins, test this please.

**Pull request merged**